lib/plugin.js

```diff
@@ -179,6 +180,68 @@ const setupProgress = (axios) => {
   axios.defaults.onDownloadProgress = onProgress
 }<% } %>
 
+<% if (options.proxyCookies) { %>
+const proxyCookies = (ctx, axios, response) => {
+  const parseSetCookies = cookies => {
+    return cookies
+      .map(cookie => cookie.split(';')[0])
+      .reduce((obj, cookie) => ({
+        ...obj,
+        ...parseCookies(cookie)
+      }), {})
+  }
+
+  const serializeCookies = cookies => {
+    return Object
+      .entries(cookies)
+      .map(([name, value]) => `${name}=${encodeURIComponent(value)}`)
+      .join('; ')
+  }
+
+  const mergeSetCookies = (oldCookies, newCookies) => {
+    const cookies = new Map()
+
+    const add = setCookie => {
+      const cookie = setCookie.split(';')[0]
+      const name = Object.keys(parseCookies(cookie))[0]
+
+      cookies.set(name, cookie)
+    }
+
+    oldCookies.forEach(add)
+    newCookies.forEach(add)
+
+    return [...cookies.values()]
+  }
+
+  const arrayify = obj => Array.isArray(obj) ? obj : [obj]
+
+  if (response.headers['set-cookie']) {
+    const setCookies = arrayify(response.headers['set-cookie'])
+
+    // Combine the cookies set on axios with the new cookies and serialize them
+    const cookie = serializeCookies({
+      ...parseCookies(axios.defaults.headers.common.cookie),
+      ...parseSetCookies(setCookies)
+    })
+
+    axios.defaults.headers.common.cookie = cookie
+
+    // If the res already has a Set-Cookie header it should be merged
+    if (ctx.res.getHeader('Set-Cookie')) {
+      const newCookies = mergeSetCookies(
```

Comment:
It is fine by HTTP to send multiple `Set-Cookie` headers and it would be a much smaller implementation.

> By your comment:  If you make multiple requests that all overwrite a certain cookie, that could increase the header size quite a bit

Yes it would but backend shouldn't send multiple `Set-Cookies` if it does, we should transparently pass them to the client like how it works on client (if each response does a `Set-Cookie` browser applies separately and in order)

If you are getting this behavior by your implementation, probably reason is during single SSR lifecycle, after we get first `Set-Cookie`, should pass it to next SSR API-calls :)

Reply:
> Yes it would but backend shouldn't send multiple Set-Cookies if it does, we should transparently pass them to the client like how it works on client (if each response does a Set-Cookie browser applies separately and in order)

The issue isn't with a single request to the backend sending duplicate `Set-Cookie` headers, but with multiple separate requests which all receive a cookie.

Consider you have the route `/cookie` which returns a header `Set-Cookie: count=i` where `i` starts at 0 and for each subsequent request count is incremented in the cookie response. The first request you get `count=0`, second `count=1`, third `count=2`. And then the following code.
```js
async asyncData({ $axios }) {
	await $axios.$get('/cookie'); // Set-Cookie: count=0
	await $axios.$get('/cookie'); // Set-Cookie: count=1
	await $axios.$get('/cookie'); // Set-Cookie: count=2
}
```

When you're in a browser, you'll end up with `count=2`, because each request is done one at a time.

When you're on the server and the `Set-Cookie` header isn't merged, the client will receive all the headers at once. It's then left up to the browser to set the correct cookie, which should be the last one, but that can't be guaranteed.
```
Set-Cookie: count=0
Set-Cookie: count=1
Set-Cookie: count=2
```

If the `Set-Cookie` headers are merged, you end up with the following response, and the browser should end up with the correct cookie.
```
Set-Cookie: count=2
```

Reply:
I know what you mean :) but it comes with the cost of much more parse/seralize logic and a library import and Something like a `Set-Cookie: count` should not honestly happen by the backend. Also, request ordering may affect which Set-Cookie finally takes over and it is hidden by the client then.

Reply:
If you're fine with edge cases not being covered that's fine by me. I think in most cases there won't be any issues if `Set-Cookie` headers aren't merged.

The cookie library is also used to parse the cookies and store them on the Axios instance. This is necessary when dealing, for example, with access tokens that get refreshed after a request, and the next request requires the new token.

Ideally, all this logic is only included in the server-side bundle, but I don't know whether that is possible here.